workflow/workflow.go

```diff
@@ -0,0 +1,92 @@
+package workflow
+
+import (
+	"fmt"
+
+	"github.com/mesg-foundation/engine/event"
+	"github.com/mesg-foundation/engine/hash"
+	eventsdk "github.com/mesg-foundation/engine/sdk/event"
+	executionsdk "github.com/mesg-foundation/engine/sdk/execution"
+)
+
+// Workflow exposes functions of the workflow
+type Workflow struct {
+	event       *eventsdk.Event
+	eventStream *eventsdk.Listener
+
+	execution *executionsdk.Execution
+
+	ErrC chan error
+}
+
+// New creates a new Workflow instance
+func New(event *eventsdk.Event, execution *executionsdk.Execution) *Workflow {
+	return &Workflow{
+		event:     event,
+		execution: execution,
+		ErrC:      make(chan error),
+	}
+}
+
+// Start the workflow engine
+func (w *Workflow) Start() error {
+	if w.eventStream != nil {
+		return fmt.Errorf("workflow engine already running")
+	}
+	w.eventStream = w.event.GetStream(nil)
+	for {
```

Comment:
```suggestion
	for event := range w.eventStream.C {
```

+ remove event below